### DISCUSSION TITLE
Uncommon mask input

### DISCUSSION THREAD

**Discussion by june94** [open-mmlab/mmsegmentation#2498] [General]
1) As I understood it right, input mask (for training) should have labels/classes as pixels, for example, 0-6 pixel range if i have 6 classes + background. 
2) Occasionally, my custom dataset has masks labeled with ade20k palette. But without any specifications about it (whole pipeline is rather simple), training doesn't crash and also labeling seems to be right.

Questions:
Am I right about 1)? Is 2) behaviour normal or it is just a coincidence? Where could I find the method where gt_semantic_seg pixels are converted to class labels if there any? 
(inspection of LoadAnnotations(), forward, loss calculations and hooks does nothing in my case)

Thank you in advance!

**Comment by MengzhangLI** [comment]
(1) In most cases, Yes. If your label 0 in annotation is background but meta info is not, like [ADE20K](https://github.com/open-mmlab/mmsegmentation/blob/master/mmseg/datasets/ade.py#L22). You should use `reduce_zero_label`, here is [doc](https://github.com/open-mmlab/mmsegmentation/blob/master/docs/en/faq.md#what-does-reduce_zero_label-work-for).

(2) Check out your customized dataset class in `./mmseg/datasets/xxx.py`, you should define class and palette information like [this](https://github.com/open-mmlab/mmsegmentation/blob/master/mmseg/datasets/chase_db1.py#L17-L19). If you want to debug, you could set breakpoint in [`custom.py`](https://github.com/open-mmlab/mmsegmentation/blob/master/mmseg/datasets/custom.py#L106) to see what's going on.

Best,

**Comment by june94** [comment]
Thank you! 
I've already debugged this part but unfortunately could not find answer for my question. 
I also found that PALETTE is only used for output/plot/show result, but not for mapping "uncommon" masks to labels/classes 0-N range (for example, in LoadAnnotations or loss calculations).
Could you please show the part where this mapping happend?
